app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -348,6 +347,66 @@ private void initViews() {
         setUpMenu();
     }
 
+    private void onNewCreate() {
+        Settings.EventHistory history = Settings.getInstance(this).getEventHistory();
+
+        boolean didShowRateDialog = history.didHappened(Settings.Event.ShowRateAppDialog);
+        boolean didShowShareDialog = history.didHappened(Settings.Event.ShowShareAppDialog);
+        boolean didPostSurvey = history.didHappened(Settings.Event.PostSurveyNotification);
+
+        if (!didShowRateDialog || !didShowShareDialog || !didPostSurvey) {
+            history.setHappened(Settings.Event.AppCreate);
+        }
+        int appCreateCount = history.getCount(Settings.Event.AppCreate);
+
+        if (!didShowRateDialog && appCreateCount >= DialogUtils.APP_CREATE_THRESHOLD_FOR_RATE_APP) {
+            DialogUtils.showRateAppDialog(this);
+            TelemetryWrapper.showFeedbackDialog();
+
+        } else if (!didShowShareDialog && appCreateCount >= DialogUtils.APP_CREATE_THRESHOLD_FOR_SHARE_APP) {
+            DialogUtils.showShareAppDialog(this);
+            TelemetryWrapper.showPromoteShareDialog();
+
+        }
+
+        if (appCreateCount >= 3 && !didPostSurvey) {
+            postSurveyNotification();
+            history.setHappened(Settings.Event.PostSurveyNotification);
+        }
+    }
+
+    private void postSurveyNotification() {
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        // TODO: Update survey url
+        intent.setData(Uri.parse("https://play.google.com/store/apps/details?id=org.mozilla.rocket"));
+        intent.setClassName(this, MainActivity.class.getName());
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        intent.putExtra(IntentUtils.EXTRA_IS_INTERNAL_REQUEST, true);
+
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, 0);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this)
+                .setSmallIcon(R.drawable.ic_notification)
+                .setContentTitle(getString(R.string.survey_notification_title, "\uD83D\uDE4C"))
+                .setContentText(getString(R.string.survey_notification_description))
+                .setStyle(new NotificationCompat.BigTextStyle().bigText((getString(R.string.survey_notification_description))))
+                .setAutoCancel(true)
+                .setContentIntent(pendingIntent)
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
+                .setVibrate(new long[0]);
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
+            builder.setColor(Color.parseColor("#00c8d7"));
+
+        } else {
+            builder.setColor(Color.parseColor("#0060df"));
+            builder.setLargeIcon(BitmapFactory.decodeResource(getResources(), R.mipmap.ic_launcher));
+            builder.setShowWhen(false);
+        }
+
+        NotificationUtil.sendNotification(this, NotificationId.SURVEY_ON_3RD_LAUNCH, builder);
```

Comment:
File another bug to remember to add NotificationChannel when targetSdk is 26 above.